lib/ocpp/v201/charge_point.cpp

```diff
@@ -3207,6 +3232,97 @@ void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest
     this->send<SetChargingProfileResponse>(call_result);
 }
 
+void ChargePoint::handle_clear_charging_profile_req(Call<ClearChargingProfileRequest> call) {
+    EVLOG_debug << "Received ClearChargingProfileRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    const auto msg = call.msg;
+    ClearChargingProfileResponse response;
+    response.status = ClearChargingProfileStatusEnum::Unknown;
+
+    // K10.FR.06
+    if (msg.chargingProfileCriteria.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.value() ==
+            ChargingProfilePurposeEnum::ChargingStationExternalConstraints) {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "InvalidValue";
+        response.statusInfo->additionalInfo = "ChargingStationExternalConstraintsInClearChargingProfileRequest";
+        EVLOG_debug << "Rejecting SetChargingProfileRequest:\n reasonCode: " << response.statusInfo->reasonCode.get()
+                    << "\nadditionalInfo: " << response.statusInfo->additionalInfo->get();
+    } else {
+        response = this->smart_charging_handler->clear_profiles(msg);
```

Comment:
I am a bit worried about the responsibilities of the smart charging handler. We filter out some edge cases but then still hand over the whole message to `clear_profiles` and `get_reported_profiles`. I think we should either move the whole handling of messages there or put an in between layer that handles the full message here. This feels like a kind of halfway solution.

Reply:
For both of these I probably missed it earlier then or it was not as apparent. Maybe we don't need to do anything with it now but we are increasing coupling this way so certainly something to look out for later.